### PR TITLE
refactor: change syntax to svelte 5

### PR DESCRIPTION
**Pull request by Sparkier**
Changes the syntax of all svelte files to Svelte 5.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -5973,6 +5973,14 @@ svelte-preprocess@^6.0.3:
   resolved "https://registry.yarnpkg.com/svelte-preprocess/-/svelte-preprocess-6.0.3.tgz#fdc1f9dc41b6f22bf8b1f059e9f21eaaae181eeb"
   integrity sha512-PLG2k05qHdhmRG7zR/dyo5qKvakhm8IJ+hD2eFRQmMLHp7X3eJnjeupUtvuRpbNiF31RjVw45W+abDwHEmP5OA==
 
+svelte-vega@^1.2.0:
```
Comment (domoritz) [inline yarn.lock:5976]: That's weird. Probably shouldn't be here.

File: packages/svelte-vega-sample/src/App.svelte [hunk 2]
```diff
@@ -15,8 +15,8 @@
 		]
 	};
 
-	let viewVL: View;
-	let specVL: VisualizationSpec = {
+	let viewVL: View | undefined = $state(undefined);
```
Comment (domoritz) [inline packages/svelte-vega-sample/src/App.svelte:18]: Sweet. Some runes.

**Review by domoritz — COMMENTED** [review]
Thank you. Should we remove the class compatibility mode I added recently?

File: yarn.lock (hunk 1 above)
Comment (Sparkier) [reply yarn.lock:5976]: hmm thats's what I get when I run `yarn` on the monorepo.

Comment (domoritz) [reply yarn.lock:5976]: Seems to indicate that we don't use the dependency from this repo somehow.

**Review by domoritz — COMMENTED** [review]
I think we also need to change the dependency to only allow svelte 5 now. So this is a breaking change.

**Comment by domoritz** [comment]
This needs to be fixed before we can merge. 

Screenshot 2024-11-11 at 10 57 08

**Comment by Sparkier** [comment]
> This needs to be fixed before we can merge.
> 
> Screenshot 2024-11-11 at 10 57 08

Done

**Comment by domoritz** [comment]
Thanks. `yarn sample:sveltekit` still doesn't work but that wasn't working before either.

**Pull request merged**